### DISCUSSION TITLE
[Bug]: Cannot run via Docker: "no configuration file provided: not found"

### DISCUSSION THREAD

**Discussion by Peter-developer01** [danny-avila/LibreChat#2754] [Troubleshooting] [answered]
### What happened?

I can't run the program via Docker. It gives the following error:

```text
no configuration file provided: not found
```

### Steps to Reproduce

1. clone this repository
2. navigate to the LibreChat directory
3. create a .env file from .env.example
4. run `docker compose up -d` and see the error

### Screenshots

![screenshot of the steps](https://github.com/danny-avila/LibreChat/assets/101456132/82e1d3e0-8f0f-4915-b144-a8e3b527f435)

Sorry for the Russian text, my system's default language is Russian.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by fuegovic** [comment]
can you share the content of your librechat folder?
```sh
ls
```

**Comment by Peter-developer01** [comment]
> can you share the content of your librechat folder?
> 
> ```shell
> ls
> ```

```bash
user@hp:/Projects/LibreChat$ ls
api        config                               docker-compose.yml  e2e                     LICENSE            packages            README.md
bun.lockb  deploy-compose.yml                   Dockerfile          index.html              package.json       prettier.config.js  utils
client     docker-compose.override.yml.example  Dockerfile.multi    librechat.example.yaml  package-lock.json  rag.yml
user@hp:/Projects/LibreChat$ 
```

**Reply by fuegovic** [reply]
What is your docker version and compose version?
```sh
docker --version
```

```sh
docker compose version
```

**Reply by Peter-developer01** [reply]
```shell
user@hp:/Projects/LibreChat$ docker --version
Docker version 24.0.5, build ced0996
user@hp:/Projects/LibreChat$ docker compose version
Docker Compose version v2.20.3
```

**Reply by fuegovic** [reply]
Try manually specifying the compose file path
```
docker compose -f docker-compose.yml logs
```

If that doesn't work you can also try to use `docker-compose` instead of `docker compose`

**Reply by Peter-developer01** [reply]
@fuegovic 

```shell
user@hp:/Projects/LibreChat$ docker compose -f docker-compose.yml logs
stat /var/lib/snapd/void/docker-compose.yml: no such file or directory
user@hp:/Projects/LibreChat$ docker-compose -f docker-compose.yml logs
stat /var/lib/snapd/void/docker-compose.yml: no such file or directory
```

Tried both `docker compose` and `docker-compose` with extra `-` added...

**Comment by spinagon** [comment, marked as answer]
https://stackoverflow.com/a/69684470/14276600

> That is a "caveat" of docker installed with snap. From docker-snap [README](https://github.com/docker-snap/docker-snap/blob/main/README.md#usage): "All files that docker needs access to should live within your $HOME folder". So that is expected behavior.

> Move your project to $HOME folder. For example, I had my project in /usr/local/src/my_project and had to move to ~/some_folder/my_project and then could run docker-compose.

So you can either do this, or install docker with apt

**Reply by fuegovic** [reply]
👆 indeed, try moving to the home folder

**Reply by Peter-developer01** [reply]
Thanks, it finally works. :)
